### PR TITLE
Fix #9772 - SONAR: code quality for 13 release

### LINKED ISSUE #9772: SONAR: code quality for 13 release
### Description

The codebase might need some attention since last time we focused on this, on this day:
- 131 critical
- 87 major

More details here: https://sonarcloud.io/project/issues?resolved=false&id=org.primefaces%3Aprimefaces

I see a lot of `Make "XXX" transient or serializable"` kind of errors, and I think it'd be great to know once and for all what to do with these warnings?

The rest should be quite easy to fix. If you have any questions, suggestions, feel free to share 😉  

### Describe the solution you would like

See how we can split the work (it's redundant work and not so fun 😅 )

### Additional context

Also I see we have some singletons for DT/TT features, I'm planning to remove those for many reasons (essentially it's an anti-pattern, and we don't need to define them as singleton)

### PR DESCRIPTION
**Pull request by Rapster**

### CODE REVIEW

**Comment by melloware** [comment]
Just a few checkstyle issues:

```
Error:  src/main/java/org/primefaces/behavior/base/AbstractBehaviorHandler.java:[56,5] (coding) DeclarationOrder: Variable access definition in wrong order.
Error:  src/main/java/org/primefaces/behavior/base/AbstractBehaviorHandler.java:[57,5] (coding) DeclarationOrder: Variable access definition in wrong order.
Error:  src/main/java/org/primefaces/behavior/base/AbstractBehaviorHandler.java:[58,5] (coding) DeclarationOrder: Variable access definition in wrong order.
Error:  src/main/java/org/primefaces/behavior/base/AbstractBehaviorHandler.java:[59,5] (coding) DeclarationOrder: Variable access definition in wrong order.
Error:  src/main/java/org/primefaces/behavior/base/AbstractBehaviorHandler.java:[67] (sizes) LineLength: Line is longer than 160 characters (found 166).
```

**Comment by tandraschko** [comment]
Will review on monday

File: primefaces/src/main/java/org/primefaces/renderkit/OutcomeTargetRenderer.java [hunk 1]
```diff
@@ -164,7 +161,7 @@ protected String getTargetURL(FacesContext context, UIOutcomeTarget outcomeTarge
                 clientWindow = context.getExternalContext().getClientWindow();
 
                 if (clientWindow != null) {
-                    clientWindowRenderingModeEnabled = ((ClientWindow) clientWindow).isClientWindowRenderModeEnabled(context);
+                    boolean clientWindowRenderingModeEnabled = ((ClientWindow) clientWindow).isClientWindowRenderModeEnabled(context);
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/renderkit/OutcomeTargetRenderer.java]: i think this break nested enabled/disabled, therefore i saved the old value

Comment (Rapster) [reply primefaces/src/main/java/org/primefaces/renderkit/OutcomeTargetRenderer.java]: Ok I see, I reverted but removed clientWindow unecessary check (just SONAR like pointed out)

**Comment by tuerker** [comment]
dear @Rapster this PR breaks my application. After this PR, i am getting the following error:

```
java.lang.NoSuchMethodError: org.primefaces.util.ComponentTraversalUtils.closestForm(Ljavax/faces/context/FacesContext;Ljavax/faces/component/UIComponent;)Ljavax/faces/component/UIForm;
	at org.primefaces.extensions.component.timer.TimerRenderer.encodeScript(TimerRenderer.java:94)
	at org.primefaces.extensions.component.timer.TimerRenderer.encodeEnd(TimerRenderer.java:75)
	at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:595)
	at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:286)
	at com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:90)
	at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1647)
	at javax.faces.render.Renderer.encodeChildren(Renderer.java:152)
	at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
	at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:278)
	at com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:90)
	at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
	at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:278)
	at com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:90)
	at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:566)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1647)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1650)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1650)
	at com.sun.faces.application.view.FaceletViewHandlingStrategy.renderView(FaceletViewHandlingStrategy.java:468)
	at com.sun.faces.application.view.MultiViewHandler.renderView(MultiViewHandler.java:170)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:132)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:132)
	at org.omnifaces.viewhandler.OmniViewHandler.renderView(OmniViewHandler.java:151)
	at com.sun.faces.lifecycle.RenderResponsePhase.execute(RenderResponsePhase.java:102)
```

It looks like that your revisions affect some primefaces extensions, namely at least the timer.
This error goes back to your change of the method

ComponentTraversalUtils.closestForm

where the argument context was dropped.

Would be great if you could do the needed adaptations also in the affected PF extensions code.

Many thanks in advance.

**Comment by Rapster** [comment]
I guess you'll have to wait for a new release of PF so @melloware can make a new release of PFE (till there don't use SNAPSHOT, or if you start using PF SNAPSHOT therefore you have to use a PFE SNAPSHOT as well)

**Comment by melloware** [comment]
@tandraschko just fixed it by putting the method back and marking it @deprecated so it will be fixed in PFE 13

**Comment by tuerker** [comment]
@melloware great this will do the job. many thanks as always.

**Pull request merged**